In [1]:
import numpy as np
import pandas as pd

In [4]:
#Sample labels
labelsamples= ["b1p1_m05","b1p1_m10","b1p1_m20","b1p1_m40","b1p5_m05","b1p5_m10","b1p5_m20","b1p5_m40","b10_m05","b10_m10","b10_m20","b10_m40"]

path='../CLs/CLs_final/'
#txt files with signal strength, expected background and signal efficiency for each nuclear configuration model
infiles = ['Eff_Bkg_index_00a.txt', 'Eff_Bkg_index_01a.txt', 'Eff_Bkg_index_02a.txt', 'Eff_Bkg_index_00b.txt', 'Eff_Bkg_index_01b.txt', 'Eff_Bkg_index_02b.txt']

#function to read the list of txt files
def readtxt(infile_array):
    listloaded = [] 
    for infile in infile_array:
        listloaded.append(np.loadtxt(path+infile, usecols=(3,4), skiprows=1))
    return listloaded

In [5]:
optimals = readtxt(infiles) 

## 1: (b_cv - b_i)/b_cv ,   2: (eff_cv - eff_i)/eff_cv ##

In [6]:
def return_shift(value):
    if value < 1:
        return -1+value
    if value > 1:
        return value-1
    if value == 1:
        return value-1

In [7]:
optimals = np.array(optimals)
optimals

array([[[1.500000e-02, 2.467099e+03],
        [1.800000e-02, 2.469529e+03],
        [1.800000e-02, 2.467099e+03],
        [1.900000e-02, 2.469529e+03],
        [1.130000e-01, 1.712385e+03],
        [1.150000e-01, 1.712385e+03],
        [1.180000e-01, 1.712385e+03],
        [1.180000e-01, 1.712385e+03],
        [3.740000e-01, 2.652313e+03],
        [3.880000e-01, 2.680509e+03],
        [3.520000e-01, 1.260043e+03],
        [3.740000e-01, 2.316885e+03]],

       [[1.400000e-02, 2.496994e+03],
        [1.600000e-02, 2.499484e+03],
        [1.700000e-02, 2.496994e+03],
        [1.800000e-02, 2.499484e+03],
        [1.160000e-01, 1.729634e+03],
        [1.170000e-01, 1.729634e+03],
        [1.200000e-01, 1.729634e+03],
        [1.200000e-01, 1.729634e+03],
        [3.740000e-01, 2.728076e+03],
        [3.870000e-01, 2.755825e+03],
        [0.000000e+00, 0.000000e+00],
        [2.660000e-01, 1.267036e+03]],

       [[2.300000e-02, 2.678891e+03],
        [2.600000e-02, 2.681152e+03],
        

In [8]:
# Vectorize the function
vectorized_nuclear_shift = np.vectorize(return_shift)

In [9]:
def color_negative_red(val):
    """
    Takes a scalar and returns a string with
    the css property `'color: red'` for negative
    strings, black otherwise.
    """
    color = 'blue' if val > 0 else 'black'
    return 'color: % s' % color

models = [vectorized_nuclear_shift(optimals[i][:][:]/optimals[0][:][:]) for i in range(6) ] 
df_models = [pd.DataFrame({'Efficiency Shift': model[:, 0]*100, 'Background Shift': model[:, 1]*100}) for model in models]


    
    
    

In [10]:
df_models[2]

Efficiency Shift  Background Shift
0          53.333333          8.584658
1          44.444444          8.569367
2          50.000000          8.584658
3          47.368421          8.569367
4          -0.884956          6.648213
5           1.739130          6.648213
6           1.694915          6.648213
7           3.389831          6.648213
8           0.267380          5.049065
9           1.288660          4.914776
10        -24.715909          2.982993
11          0.000000          5.417403

In [22]:
df_models[1]

Efficiency Shift  Background Shift
0          -6.666667          1.211747
1         -11.111111          1.212984
2          -5.555556          1.211747
3          -5.263158          1.212984
4           2.654867          1.007309
5           1.739130          1.007309
6           1.694915          1.007309
7           1.694915          1.007309
8           0.000000          2.856488
9          -0.257732          2.809765
10       -100.000000       -100.000000
11        -28.877005        -45.312953

In [15]:
df_models[0].to_latex(index=False)

/home/leoperes/.local/lib/python3.12/site-packages/matplotlib/projections/__init__.py:63: UserWarning: Unable to import Axes3D. This may be due to multiple versions of Matplotlib being installed (e.g. as a system package and as a pip package). As a result, the 3D projection is not available.
  warnings.warn("Unable to import Axes3D. This may be due to multiple versions of "


'\\begin{tabular}{rr}\n\\toprule\nEfficiency Shift & Background Shift \\\\\n\\midrule\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n\\bottomrule\n\\end{tabular}\n'

In [23]:
print(df_models[5].to_latex(index=False))

\begin{tabular}{rr}
\toprule
Efficiency Shift & Background Shift \\
\midrule
120.000000 & 19.380818 \\
94.444444 & 19.370980 \\
88.888889 & 19.370980 \\
78.947368 & 19.357600 \\
2.654867 & 16.388703 \\
2.608696 & 16.388703 \\
5.932203 & 16.388703 \\
5.084746 & 16.388703 \\
-2.680965 & 12.369804 \\
-1.033592 & 12.259240 \\
-24.129353 & 11.225686 \\
-1.807229 & 11.584365 \\
\bottomrule
\end{tabular}



&$m_\chi$ (GeV)}& hA\_BR & hA\_LFG & hA\_ESF  & hN\_BR  & hN\_LFG  & hN\_ESF  \\
\hline
\multirow{4}{*}{$\gamma = 1.1$}& 5 &0.015 & -6.6\%  & +53\%  &  &  & \\
  &10& 0.018&-11.1\%& +44\%  &  &  &   \\
  &20&0.018 &0\%  & +54\% & +55\%  &  &   \\
  &40&0.019 & -5.2\%&  &  & +47% &   \\
\hline
\multirow{4}{*}{$\gamma = 1.5$}& 5 &0.113  &+2.6\%  &  &  &  & \\
  &10& 0.115& 1.73\% &  &  &  &   \\
  &20& 0.118& 1.69\% &  &  &  &   \\
  &40& 0.118& 1.69\%  &  &  &  &   \\
  \hline
\multirow{4}{*}{$\gamma = 10$}& 5 & 0.373 &0.0\%  &  &  &  & \\
  &10&0.387 & -0.25\%&  &  &  &   \\
  &20& 0.402& X &  &  &  &   \\
  &40& 0.332& -4.8\%&  &  &  &   \\
\hline


53.333333
44.444444
55.555556
47.368421
-0.884956
1.739130
1.694915
3.389831
0.268097
1.033592
-21.890547
0.602410

 & 11.153310 \\
 & 11.136119 \\
 & 11.136119 \\
 & 11.120641 \\
 & 10.428919 \\
 & 10.428919 \\
 & 10.428919 \\
 & 10.428919 \\
 & 7.396272 \\
 & 7.293977 \\
 & 5.610628 \\
 & 6.409939 \\
